src/ocl/tensorocl.cpp

```diff
@@ -174,8 +174,24 @@ void OpTensor3d(const Handle& handle,
 
     std::string network_config{};
 
-    network_config = std::to_string(bTensorDesc.GetType()) + std::to_string(aTensorDesc.GetType()) +
-                     std::to_string(tensorOp);
+    network_config = std::to_string(bTensorDesc.GetType()) + "-" +
+                     std::to_string(aTensorDesc.GetType()) + "-" + std::to_string(tensorOp) + "-";
+
+    // for naive tensor ops
+    size_t RD_BLCK              = (clens[2] % 4 == 0) ? 4 : (clens[2] % 2 == 0) ? 2 : 1;
+    const std::string data_type = GetDataType(bTensorDesc.GetType());
+    const std::string READ_TYPE = (RD_BLCK == 1) ? data_type : data_type + std::to_string(RD_BLCK);
+
+    size_t total_work = std::max(size_t(clens[2] / RD_BLCK), size_t(1));
```

Comment:
you don't need size_t cast operator for literal 1. Secondly, since we are using c++, for any casting operator us static_cast<size_t>(clen[2]/RD) rather than C-style.

Reply:
Will do